app/src/main/java/com/example/henzoshimada/feeltrip/Mood.java

```diff
@@ -139,51 +158,70 @@ public String getEmotionalState() {
     }
 
     public void setEmotionalState(String emotionalState) {
-        this.emotionalState = emotionalState;
-        setStateByIndex(0);
+        if (!getEmotionalState().equals(emotionalState)) {
+            this.emotionalState = emotionalState;
+            setStateByIndex(0);
+        }
     }
 
     public void setImage(String image) {
-        this.image = image;
-        setStateByIndex(5);
+        if (getImage() == null) {
+            this.image = TextUtils.htmlEncode(image);
+            setStateByIndex(5);
+        } else if (!getImage().equals(image)) {
+            this.image = TextUtils.htmlEncode(image);
```

Comment:
HTML encode our image String before we send it to ElasticSearchController, allowing us to call it properly from our edit query